tests/llmcompressor/transformers/compression/test_decompress.py

```diff
@@ -0,0 +1,128 @@
+import copy
+import shutil
+import tempfile
+import unittest
+
+import torch
+from compressed_tensors import QUANTIZATION_CONFIG_NAME
+from compressed_tensors.compressors import ModelCompressor
+from compressed_tensors.quantization import QuantizationStatus
+from parameterized import parameterized_class
+from transformers import AutoConfig, AutoModelForCausalLM, AutoTokenizer
+from transformers.utils.quantization_config import CompressedTensorsConfig
+
+from tests.testing_utils import parse_params, requires_gpu
+
+CONFIG_DIR = "tests/llmcompressor/transformers/compression/decompression_configs"
+
+
+@requires_gpu
+@parameterized_class(parse_params(CONFIG_DIR))
+class TestDecompression(unittest.TestCase):
+    """
+    Check that HFQuantizer decompression is working as expected.
+    Manually decompress a compressed model and compare the generations
+
+    Decompression:
+    Given a skeleton model and path to the optimized model,
+    write the optimized model's safetensors to the skeleton model and decompress
+    Ex. write weight_scale to the skeleton model and then convert from fp4 to fp16
+
+    """
+
+    compressed_model_stub = None
+    skeleton_model_stub = None
+
+    SAMPLE_INPUTS = [
+        "I love 4-bit quantization because",
+        "What is the capital of France?",
+        "def fibonacci(n):",
+    ]
+
+    @classmethod
+    def setUpClass(self):
+        self.test_dir = tempfile.mkdtemp()
+        self.tokenizer = AutoTokenizer.from_pretrained(self.compressed_model_stub)
+
+        # Decompress using HFQuantizer from AutoModelForCausalLM
+        self.decompressed_model_hf_quantizer = AutoModelForCausalLM.from_pretrained(
+            self.compressed_model_stub,
+            torch_dtype="auto",
+            device_map="auto",
+            quantization_config=CompressedTensorsConfig(run_compressed=False),
+        )
+
+        # Manually decompress this model
+        self.dense_model = AutoModelForCausalLM.from_pretrained(
+            self.skeleton_model_stub,
+            torch_dtype=self.decompressed_model_hf_quantizer.dtype,
+            device_map=self.decompressed_model_hf_quantizer.device,
+        )
+
+        assert not hasattr(
```

Comment:
Should we not also check `decompressed_model_hf_quantizer`?

Reply:
Sure, dense model wont have it so a check for that